### PR TITLE
DolphinQt: Rework TAS input threading

### PR DESCRIPTION
**Pull request by JosJuice**
This gets rid of a blocking operation, improving performance and fixing https://bugs.dolphin-emu.org/issues/12893.

This also makes us no longer directly access the state of certain UI elements from the CPU thread, which probably wasn't thread-safe but doesn't seem to have caused any observable issues so far.

This is an alternative to PR #11620.

### CODE REVIEW

**Comment by JosJuice** [comment]
This PR isn't quite finished yet – the new approach is fully implemented for buttons, but hasn't been implemented for analog controls yet. I've opened the PR early to get feedback on whether this is a good approach.

**Comment by Dentomologist** [comment]
The threading model seems fine to me.

On master, holding a controller button and unchecking the box in the TAS window will result in the controller input and checkbox staying pressed. In this PR, the checkbox (and actual input sent to the game) will become unpressed. Is that change intentional?

**Comment by JosJuice** [comment]
Yes, though the main reason I did it was so it would be consistent with how analog inputs worked so I could simplify the code.

File: Source/Core/DolphinQt/TAS/TASControlState.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  const State new_state{static_cast<u16>(cpu_thread_state.version + 1), new_value};
+  m_cpu_thread_state.store(new_state, std::memory_order_relaxed);
+
+  return true;
+}
+
+void TASControlState::OnUIValueChanged(u16 new_value)
+{
+  const State ui_thread_state = m_ui_thread_state;
+  const State cpu_thread_state = m_cpu_thread_state.load(std::memory_order_relaxed);
```
Comment (AdmiralCurtiss) [inline Source/Core/DolphinQt/TAS/TASControlState.cpp:38]: Why are you loading the CPU state here?

**Pull request merged**